hpcflow/sdk/helper/helper.py

```diff
@@ -136,6 +139,54 @@ def start_helper(
             sys.exit(1)
 
 
+def modify_helper(
+    app,
+    timeout=DEFAULT_TIMEOUT,
+    timeout_check_interval=DEFAULT_TIMEOUT_CHECK,
+    watch_interval=DEFAULT_WATCH_INTERVAL,
+):
+    PID_file = get_PID_file_path(app)
+    if PID_file.is_file():
+        [helper_pid, to, tci, wi] = strip_helper_PID(app)
+        if (
+            float(to) != timeout
+            or float(tci) != timeout_check_interval
+            or float(wi) != watch_interval
+        ):
+            logger = get_helper_logger(app)
+            logger.info(
+                f"Modifying helper with pid={helper_pid}"
+                f" to: timeout={timeout!r}, timeout_check_interval="
+                f"{timeout_check_interval!r} and watch_interval={watch_interval!r}."
+            )
+
+            if isinstance(timeout, timedelta):
+                timeout = timeout.total_seconds()
+            if isinstance(timeout_check_interval, timedelta):
+                timeout_check_interval = timeout_check_interval.total_seconds()
+            if isinstance(watch_interval, timedelta):
+                watch_interval = watch_interval.total_seconds()
+
+            try:
+                with PID_file.open("wt") as fp:
+                    fp.write(f"pid = {helper_pid}\n")
+                with PID_file.open("at") as fp:
```

Comment:
Not sure we need this line?

Reply:
The reason I have 171 and 173 is that 171 overwrites the file (so it deletes previous content) and 173 appends to it. But if you think there's a better way to do this, I'd be happy to change it.

Reply:
[e0569e4](https://github.com/hpcflow/hpcflow-new/pull/340/commits/e0569e4048b9228aea4504b4bb61e6e73569b5ef)